azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -254,7 +495,10 @@ func resourceArmEventGridEventSubscriptionCreateUpdate(d *schema.ResourceData, m
 		return fmt.Errorf("One of the following endpoint types must be specificed to create an EventGrid Event Subscription: %q", getEnpointTypes())
 	}
 
-	filter := expandEventGridEventSubscriptionFilter(d)
+	filter, err := expandEventGridEventSubscriptionFilter(d)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating EventGrid Event Subscription %q (Scope %q): %s", name, scope, err)
```

Comment:
in this instance since this error is expanding the filter, we can raise this via:

```suggestion
		return fmt.Errorf("expanding filters for EventGrid Event Subscription %q (Scope %q): %+v", name, scope, err)
```